app/src/main/java/com/epfl/drawyourpath/authentication/MockAuth.kt

```diff
@@ -36,11 +38,20 @@ class MockAuth(
 
                 return false
             }
+
+             override fun updatePassword(password: String): CompletableFuture<Void> {
+                 if (password.isEmpty() || password.length < 5) {
+                     val res = CompletableFuture<Void>()
+                     res.completeExceptionally(Exception("Password is empty"))
```

Comment:
Suggestion: Extract the password validation logic into a separate function for better maintainability. Perhaps in the future, more checks should be added to ensure secure passwords.

Reply:
It is the `MockAuth` for tests, Firebase handles security checks automatically for the actual Auth object. I've added this test to emulate Firebase policy for testing, but it is a lot less restrictive. Do you still think it would be worth it to improve password security for the mock?